MetaMask/metamask-extension
# Removing unused deprecated `Text` component

## Explanation
Finally part of the updating deprecated `Text` paths in the extension is to remove the deprecated component as it is no longer in use.

* Fixes #19986

## Screenshots/Screencaps
No results for `text/deprecated`

![Screenshot 2023-07-20 at 10 51 58 AM](https://github.com/MetaMask/metamask-extension/assets/8112138/4790246a-8e16-4cb3-949e-ed19afd5f497)


## Manual Testing Steps

- Pull this branch
- Search part of the import path which would return a result if the deprecated component was used `text/deprecated` 
- See no results

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
